wrapperjakarta/src/main/java/com/genexus/ws/GXHandlerChain.java

```diff
@@ -27,13 +27,15 @@ public boolean handleMessage(SOAPMessageContext messageContext) {
         java.io.ByteArrayOutputStream out = new java.io.ByteArrayOutputStream();
         ;
         try {
-            messageContext.getMessage().writeTo(out);
-            String messageBody = new String(out.toByteArray(), "utf-8");
-            if (Boolean.FALSE.equals(outboundProperty)) {
-                messageContext.put(GX_SOAP_BODY, messageBody);
-                messageContext.setScope(GX_SOAP_BODY, MessageContext.Scope.APPLICATION);
-            }
-            logger.debug(messageBody);
+			if (logger.isDebugEnabled()) {
+				messageContext.getMessage().writeTo(out);
+				String messageBody = new String(out.toByteArray(), "utf-8");
+				if (Boolean.FALSE.equals(outboundProperty)) {
+					messageContext.put(GX_SOAP_BODY, messageBody);
```

Comment:
`messageContext.put(GX_SOAP_BODY, messageBody);` will only be executed if DEBUG is Enabled. 
Is this correct? Just to be sure.  
Ignore if not relevant.